src/validators/StringValidator.ts

```diff
@@ -29,6 +37,28 @@ export class StringValidator<T extends string> extends BaseValidator<T> {
 		return this.addConstraint(stringLengthNe(length) as IConstraint<T>);
 	}
 
+	public email(): this {
+		const emailRegex = /^(([^<>()[\]\.,;:\s@\"]+(\.[^<>()[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
+		return this.addConstraint(stringRegex(emailRegex, 'email') as IConstraint<T>);
+	}
+
+	public url(allowedProtocols: string[] = ['https?']): this {
+		const urlRegex = new RegExp(`^(?:${allowedProtocols.join('|')}):\/\/[^\s\.]+\.\\S{2,}$`);
+		return this.addConstraint(stringRegex(urlRegex, 'url') as IConstraint<T>);
+	}
+
+	public uuid(v?: 1 | 2 | 3 | 4 | 5): this {
+		const uuidRegex = new RegExp(
+			`^([0-9A-F]{8}-[0-9A-F]{4}-[${v ?? '1-5'}][0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}|00000000-0000-0000-0000-000000000000)$`,
+			'i'
+		);
```

Comment:
Are all UUID versions really the same format?

Reply:
Also, it should default to UUIDv4, since that's the most popular version.

wdyt? @favna @vladfrangu

Reply:
> Are all UUID versions really the same format?

https://stackoverflow.com/a/38191078/16893382

I've tested with few uuids and it was working fine.

Reply:
Honestly, this should use `null` if you want `1-5`, otherwise default to `4`.

Reply:
added